sql/src/main/java/io/crate/execution/engine/collect/NestableCollectExpression.java

```diff
@@ -23,39 +23,33 @@
 package io.crate.execution.engine.collect;
 
 import io.crate.expression.NestableInput;
-
 import java.util.function.Function;
 
 /**
  * Base interface for row based expressions.
  *
- * @param <TReturnValue> The returnType of the expression
+ * @param <R> The returnType of the expression
  */
-public abstract class NestableCollectExpression<TRow, TReturnValue>
-    implements CollectExpression<TRow, TReturnValue>, NestableInput<TReturnValue> {
+public interface NestableCollectExpression<T, R> extends CollectExpression<T, R>, NestableInput<R> {
```

Comment:
The changes here in this class look unrelated to the ColumnRegistry changes. Can we revert them / split them out into a separate PR or commit?

Reply:
yes, sure.